modules/service-fips.sh

```diff
@@ -25,6 +26,8 @@ fips_enable() {
     check_token "$FIPS_REPO_URL" "$token"
     apt_add_repo "$FIPS_REPO_LIST" "$FIPS_REPO_URL" "$token" \
                  "${KEYRINGS_DIR}/${FIPS_REPO_KEY_FILE}"
+    apt_add_repo_pinning "$FIPS_REPO_PREFERENCES" \
+                         LP-PPA-ubuntu-advantage-fips 1001
```

Comment:
Since this is in the `fips_enable()` path, it doesn't look like we will get this feature by merely upgrading ubuntu-advantage-fips in a fips-enabled system? People should disable (which is not possible with fips) and re-enable?

Reply:
True, but it's the same with other things (like authentication credentials in a separate auth file).
Supporting upgrades gets complicated, I would not worry about it in this case

Reply:
I guess we could support `fips-disable` (by removing the repositories as we do for esm)?

Reply:
fips-enable does more than enabling repositories unlike esm. It sets FIPS parameters kernel command line, configures boot loader etc. "fips-disable" as a feature may need to undo all that, or else the system will continue to run in fips mode.
For new users, pinning repos through fips_enable may work when this feature is made available through ubuntu-advantage tool. For existing users who already enabled fips, needs a different way to pin their repos.

Reply:
We could create the pinning config file in the package's postinst, if we detect that fips is enabled already.

Reply:
That may be a good way to do it when ubuntu-advantage package undergoes update.

Reply:
We should just probably not use `ubuntu-advantage is-fips-enabled` in a maintainer script, because that calls `apt-cache policy`, and there could be locks in play here. We would be calling that inside an apt transaction. I'm thinking a simple check for config files plus perhaps running kernel for this.

Reply:
Ok, let's leave this for another time. It's not worth handling the upgrade case, and it can be sorted via documentation and instructions.

Reply:
Let me just give this branch a run and then I'll approve it.

Reply:
@panlinux +1 on considering adding the pinning in the postints, but it seems we can defer it

Reply:
Found a bug in modules/apt.sh in master, which I fixed as a `[trivial]`. That's why I merged master back into this.